plugins/volumeFilter.go

```diff
@@ -158,6 +158,11 @@ func (f *volumeFilter) Apply(ops []txnbuild.Operation, sellingOffers []hProtocol
 }
 
 func volumeFilterFn(dailyOTB *VolumeFilterConfig, dailyTBBAccumulator *VolumeFilterConfig, op *txnbuild.ManageSellOffer, baseAsset hProtocol.Asset, quoteAsset hProtocol.Asset, lp limitParameters) (*txnbuild.ManageSellOffer, error) {
+	// if not in exact mode, return a nil op
```

Comment:
From the previous function, if the mode was not `Exact`, a `nil` op would be returned. I wasn't sure if that was a desired invariant of the function or a side-effect of its implementation. So I'm just flagging this early return to confirm correctness.

Reply:
This is not correct.

If the mode is `Ignore` it will not return `nil` but will return a non-nil op.
If it was to return `nil` then anyone who used the `Exact` feature would never place any orders as we would drop all of them before we placed it!

This demonstrates an incorrect understanding of what is going on in this PR. does not make sense to review the rest of this PR.

Reply:
My bad on this.  Ah, I tried to redo my chart and see what I did wrong here. Definitely a strong argument for this refactor.

For fuller context, I had put that we kept selling in two cases:
1. Nil cap.
2. Projected sold exceeds cap; mode is exact; and new amount is positive and changed on the operation

Of course, this misses the obvious case:
3. Projected sold is under the cap. In that case, even if the mode is `Ignore`, then `keepSellingBase`/`keepSellingQuote` would remain true.

However, I think one point of conversation in our sync yesterday might then be incorrect. We said that the mode can be checked outside the function that computes remaining capacity. If we do this, we must double compute the projected sold amount, and we lose much of the benefit of this abstraction. I'll think more about this.